File: csharp/src/impl/ReplicationMasterStorageImpl.cs
#if WITH_REPLICATION
namespace Volante.Impl
{
    using Volante;

    public class ReplicationMasterDatabaseImpl : DatabaseImpl, ReplicationMasterDatabase
    {
        public ReplicationMasterDatabaseImpl(string[] hosts, int asyncBufSize)
        {
            this.hosts = hosts;
            this.asyncBufSize = asyncBufSize;
        }

        public override void Open(IFile file, int cacheSizeInBytes)
        {
            base.Open(asyncBufSize != 0
                ? (ReplicationMasterFile)new AsyncReplicationMasterFile(this, file, asyncBufSize)
                : new ReplicationMasterFile(this, file),
                cacheSizeInBytes);
        }

        public int GetNumberOfAvailableHosts()
        {
            return ((ReplicationMasterFile)pool.file).GetNumberOfAvailableHosts();
        }

        internal string[] hosts;
        internal int asyncBufSize;
    }
}
#endif

File: java/src/org/nachodb/impl/ReplicationMasterStorageImpl.java
package org.nachodb.impl;

import java.io.*;
import java.net.*;

import org.nachodb.*;


public class ReplicationMasterStorageImpl extends StorageImpl implements ReplicationMasterStorage
{ 
    public ReplicationMasterStorageImpl(String[] hosts, int asyncBufSize) { 
        this.hosts = hosts;
        this.asyncBufSize = asyncBufSize;
    }
    
    public void open(IFile file, int pagePoolSize) {
        super.open(asyncBufSize != 0 
                   ? (ReplicationMasterFile)new AsyncReplicationMasterFile(this, file, asyncBufSize)
                   : new ReplicationMasterFile(this, file),
                   pagePoolSize);
    }

    public int getNumberOfAvailableHosts() { 
        return ((ReplicationMasterFile)pool.file).getNumberOfAvailableHosts();
    }

    String[] hosts;
    int      asyncBufSize;
}
